exec/cnex/number.c

```diff
@@ -45,7 +45,8 @@ void number_toString(Number x, char *buf, size_t len)
 
     // If we didn't find an exponent marker, then just return the buffer that it is.
     if (*s != 'E') {
-        buf = val;
+        // Number is possibly +Nan, or +Inf.
+        memcpy(buf, val, len);
```

Comment:
What happens if `len` is greater than `sizeof(val)`? That would read past the end of the buffer.